### PR TITLE
Fix rviz crash on exit

### LINKED ISSUE #73: RViz segfaults and doesn't end properly
Using the current ros2 master, you can start RViz and then end it regularly (press "Close without saving"). At that point, I see a segmentation fault message on my Ubuntu console.

On Windows (using the fixes of the current PR), after pressing "Close without saving", you get a "Program is not responding" window.

I investigated the causes:
- First of all, there is a problem in `selection_manager.cpp:128`. The line 
`vis_manager_->getSceneManager()->destroyCamera(camera_);` throws the segmentation fault, since the method getSceneManager() returns a null pointer at that point. Probably, everything is already deleted. 
- If I comment out the line, the segfault does not occur anymore, but the problem hangs and doesn't respond (this is probably also the issue Windows is facing). The stack-trace according to gdb is

```
# 0  0x00007ffff36ed98d in pthread_join (threadid=140735676413696, thread_return=0x0) at pthread_join.c:90
# 1  0x00007ffff2e01b97 in std::thread::join() () from /usr/lib/x86_64-linux-gnu/libstdc++.so.6
#2  0x00007ffff397342e in tf2_ros::TransformListener::~TransformListener() ()
   from /home/martin/gitRepos/ros2_master/install/lib/libtf2_ros.so
# 3  0x0000000000512fdc in __gnu_cxx::new_allocator<tf2_ros::TransformListener>::destroy<tf2_ros::TransformListener> (this=0x20577e0, 
    __p=0x20577e0) at /usr/include/c++/5/ext/new_allocator.h:124
# 4  0x0000000000512f6f in std::allocator_traits<std::allocator<tf2_ros::TransformListener> >::destroy<tf2_ros::TransformListener> (
    __a=..., __p=0x20577e0) at /usr/include/c++/5/bits/alloc_traits.h:542
# 5  0x0000000000512cc7 in std::_Sp_counted_ptr_inplace<tf2_ros::TransformListener, std::allocator<tf2_ros::TransformListener>, (__gnu_cxx::_Lock_policy)2>::_M_dispose (this=0x20577d0) at /usr/include/c++/5/bits/shared_ptr_base.h:531
# 6  0x000000000048b51a in std::_Sp_counted_base<(__gnu_cxx::_Lock_policy)2>::_M_release (this=0x20577d0)
    at /usr/include/c++/5/bits/shared_ptr_base.h:150
# 7  0x000000000048abc7 in std::__shared_count<(__gnu_cxx::_Lock_policy)2>::~__shared_count (this=0x252e1f0, __in_chrg=<optimized out>)
    at /usr/include/c++/5/bits/shared_ptr_base.h:659
# 8  0x00000000004b9fb6 in std::__shared_ptr<tf2_ros::TransformListener, (__gnu_cxx::_Lock_policy)2>::~__shared_ptr (this=0x252e1e8, 
    __in_chrg=<optimized out>) at /usr/include/c++/5/bits/shared_ptr_base.h:925
# 9  0x00000000004b9fee in std::shared_ptr<tf2_ros::TransformListener>::~shared_ptr (this=0x252e1e8, __in_chrg=<optimized out>)
    at /usr/include/c++/5/bits/shared_ptr.h:93
# 10 0x00000000004b6b56 in rviz_common::FrameManager::~FrameManager (this=0x252e180, __in_chrg=<optimized out>)
    at /home/martin/gitRepos/rvizPort/aos-rviz2/src/rviz_common/src/rviz_common/frame_manager.cpp:76
# 11 0x00000000004b6ba6 in rviz_common::FrameManager::~FrameManager (this=0x252e180, __in_chrg=<optimized out>)
    at /home/martin/gitRepos/rvizPort/aos-rviz2/src/rviz_common/src/rviz_common/frame_manager.cpp:78
# 12 0x0000000000514294 in rviz_common::VisualizationManager::~VisualizationManager (this=0x2536b40, __in_chrg=<optimized out>)
    at /home/martin/gitRepos/rvizPort/aos-rviz2/src/rviz_common/src/rviz_common/visualization_manager.cpp:274
# 13 0x0000000000514334 in rviz_common::VisualizationManager::~VisualizationManager (this=0x2536b40, __in_chrg=<optimized out>)
    at /home/martin/gitRepos/rvizPort/aos-rviz2/src/rviz_common/src/rviz_common/visualization_manager.cpp:281
# 14 0x0000000000500831 in rviz_common::VisualizationFrame::~VisualizationFrame (this=0x17a7400, __in_chrg=<optimized out>)
    at /home/martin/gitRepos/rvizPort/aos-rviz2/src/rviz_common/src/rviz_common/visualization_frame.cpp:173
# 15 0x00000000005009de in rviz_common::VisualizationFrame::~VisualizationFrame (this=0x17a7400, __in_chrg=<optimized out>)
    at /home/martin/gitRepos/rvizPort/aos-rviz2/src/rviz_common/src/rviz_common/visualization_frame.cpp:182
# 16 0x0000000000482df2 in rviz_common::VisualizerApp::~VisualizerApp (this=0x7fffffffd9e0, __in_chrg=<optimized out>)
    at /home/martin/gitRepos/rvizPort/aos-rviz2/src/rviz_common/src/rviz_common/visualizer_app.cpp:309
# 17 0x00000000004829e0 in main (argc=1, argv=0x7fffffffdb28) at /home/martin/gitRepos/rvizPort/aos-rviz2/src/rviz2/src/main.cpp:39
```

This might be related to the cleanup of tf2-usage we still have to do.

### PR DESCRIPTION
**Pull request by anhosi**
fixes #73

### CODE REVIEW

File: rviz_common/src/rviz_common/selection/selection_manager.cpp [hunk 1]
```diff
@@ -125,8 +125,6 @@ SelectionManager::~SelectionManager()
   }
   delete[] reinterpret_cast<uint8_t *>(depth_pixel_box_.data);
 
-  vis_manager_->getSceneManager()->destroyCamera(camera_);
```
Comment (wjwwood) [inline rviz_common/src/rviz_common/selection/selection_manager.cpp:128]: Why is this needed? Is there somewhere else where this is happening instead?

File: rviz_common/src/rviz_common/visualizer_app.cpp [hunk 2]
```diff
@@ -307,6 +309,7 @@ bool VisualizerApp::init(int argc, char ** argv)
 VisualizerApp::~VisualizerApp()
 {
   delete continue_timer_;
+  rclcpp::shutdown();
```
Comment (wjwwood) [inline rviz_common/src/rviz_common/visualizer_app.cpp]: This needs to be wrapped into the `ros_integration` namespace, see how init and ok are done and emulate that.

**Review by wjwwood — CHANGES_REQUESTED** [review]

File: rviz_common/src/rviz_common/selection/selection_manager.cpp (hunk 1 above)
Comment (anhosi) [reply rviz_common/src/rviz_common/selection/selection_manager.cpp:128]: At this point the scene manager is already deleted and `getSceneManager()` returns a null pointer. Also the OgreSceneManager does delete all cameras in its destructure. Unfortunately, the Ogre documentations does not say so, but a look at the [Ogre code](https://github.com/OGRECave/ogre/blob/ea41ed9ec36ca896b695b41bee3dbda8855f0767/OgreMain/src/OgreSceneManager.cpp#L194) is necessary. Therefore the explicit camera destruction is not necessary.

**Comment by Martin-Idel-SI** [comment]
The shutdown is currently untested. We thought about how to unit test, but didn't see an easy way, so we'll leave it for now, but we should revisit it at some point.

**Review by wjwwood — APPROVED** [review]

**Pull request merged**